### ISSUE TITLE
Picking operation is not working in Emscripten + vtkSDL2OpenGLRenderWindow

### ISSUE THREAD

**Issue by EJShim** [InsightSoftwareConsortium/itk-js#346]
I am trying several operations for newly updated VTK with emscripten rendering.

I found that [Picking Example](https://vtk.org/Wiki/VTK/Examples/Cxx/Interaction/Picking) seems not working for now.

Getting 2D coordinates works fine, but `vtkPropPicker->Pick(...)` results warning and wrong results.

```

    virtual void OnLeftButtonDown() override
    {
      int* clickPos = this->GetInteractor()->GetEventPosition();


      // Pick from this location.
      vtkSmartPointer<vtkPropPicker>  picker = vtkSmartPointer<vtkPropPicker>::New();
      picker->Pick(clickPos[0], clickPos[1], 0, this->GetDefaultRenderer());

      double* pos = picker->GetPickPosition();
      std::cout << "Pick position (world coordinates) is: " << pos[0] << " " << pos[1] << " " << pos[2] << std::endl;
      std::cout << "Picked actor: " << picker->GetActor() << std::endl;
```

The picked position in world coordinates is always 0 0 0, and the browser shows warning saying : 

```
[.WebGL-000001CEB1519F40] GL_INVALID_OPERATION: Invalid format and type combination.
```

**Comment by Anik2006** [comment]
Hi, Any update on this. I am getting similar error.
Thanks

**Comment by jspanchu** [comment]
Can you please try to re-compile against vtk/master branch? This has been recently fixed.
Let us know if you still face this problem.

**Comment by EJShim** [comment]
@jspanchu  I tried to build build vtk/master using emscripten compiler, but It fails because of [this issue ](https://gitlab.kitware.com/vtk/vtk/-/issues/18724)

I treid using master, release and v9.2.5, and they all has same issue.

**Comment by EJShim** [comment]
@jspanchu  I succeeded re-compile against vtk/release branch (c0f1da90), but the picking still does not working properly.

You can try to execute example here : [https://github.com/EJShim/emscripten-vtk-examples](https://github.com/EJShim/emscripten-vtk-examples)

**Comment by jspanchu** [comment]
> I succeeded re-compile against vtk/release branch (c0f1da90), but pickign still is not working properly.

Yes, actor picking works, however cell and point selection will not yet work with GLES. However there's a fix in the works. Once it lands in VTK master, I'll post an update on this one. Thanks for providing those examples!

> I tried to build build vtk/master using emscripten compiler, but It fails because of [this issue](https://gitlab.kitware.com/vtk/vtk/-/issues/18724)

We're in the process of updating VTK build documentation with the correct cmake configure command. There are also plans to make sure it is continuously tested on VTK CI.

**Comment by sheonny** [comment]
Vtkimageviewer2 cannot be found in the VTK built by emscripten. Have you encountered this problem?
